dmv/dmv.js

```diff
@@ -19,13 +19,18 @@ DMV.prototype.currentCustomerFor = function(agent) {
 }
 
 DMV.prototype.nextCustomer = function() {
+  if (this.customers.length === 0) {
+    throw new Error('No customers')
+  }
   const currentCustomer = this.customers.shift()
-  this.agentList.find(agent => {
+  let currentAgent = null;
+  this.agentList.forEach(agent => {
     if (!this.attending[agent]) {
       this.attending[agent] = currentCustomer
-      return true
+      currentAgent = agent
     }
   })
+  return {agent: currentAgent, customer: currentCustomer}
```

Comment:
good job explicitly returning the object on this method.....side effects are buggy....would like to see more Composition in this API

Reply:
This looks cleaner than before.